aiverify-foundation/aiverify
# Test Results Page

## Description

Changes include:
- Main listing page: Displays a list of test results including filters (by model type or algorithms), search bar (with fuzzy search), and sorting filter.
- Detailed results view: Shows detailed information for a specific test result, with tabs for algorithm arguments, and output and artifacts. Artifacts appear in modal popup after clicking. Arguments, outputs and artifacts (individually and in a zip file) can be downloaded.
- Homepage: Updated the homepage to only display Create New Project and Manage. Manage redirects to /home/manage where the other modules are under.

Components:
- TestResultslist.tsx: Handles the display of all test results in the main listing page and the split pane view.
- TestResultsDetail.tsx: Handles the display of selected test result in the right pane.
- TestResultsCard.tsx: Summarized view for each test result in the displayed list.
- SplitPane.tsx: Splits the listing page where left pane is the list of test results with filters, sorting, and search bar and right pane is the selected test result.
- ResultsNameHeader.tsx: Handles the editing of the test results name and deleting of test result.
- ResultsHeader.tsx: Page Header with burger menu, bell icon, and AIVerify logo.
- FilterButtons.tsx: Contains the filters, sort by filters and search bar with dynamic positioning based on split pane being toggled or not.
- DropdownMenu.tsx: Reusable dropdown menu for the filter and sort by filters, which is dynamically positioned and adjusts width to the longest option.
- ArtifactPopup.tsx: Handles the display of the selected artifact with close and download button for this artifact.
- ActionButton.tsx: Contains the Run New Test and Upload Results buttons.

## How to Test

1. Navigate to aiverify-portal and run `npm run dev`
2. Navigate to aiverify-apigw and run `pip install -e .` and `python -m aiverify_apigw` in virtual environment.
3. Test features:
- Check if modules under Manage redirects to correct pages.
- Apply filters (sort by, filters) to verify filtering functionality.
- Test search bar to ensure fuzzy searching works.
- Click on a test result to verify if split pane works and detailed test results is displayed. Clicking on same test result should toggle off the split pane, whereas clicking on another should update the display to new test result.
- For detailed test results view, confirm that test arguments and output artifacts display correctly in their respective tabs.
- Click on artifact to verify if display works and download button works for individual artifact and all artifacts in zip file.
- Click on download button for algorithm arguments and output to verify downloading functionality.

Known Issues/Additional Notes:
- Future work: formatting the components/making page look nicer
